workflows/prognostic_c48_run/runtime/transformers/fv3fit.py

```diff
@@ -84,7 +84,15 @@ def predict(self, inputs: State) -> State:
 
     def apply(self, prediction: State, state: State):
         if self.config.online:
-            state.update(prediction)
+            prediction_filled_nans = {}
+            for k, v in prediction.items():
+                try:
+                    existing_value = state[k]
+                except KeyError:
+                    prediction_filled_nans[k] = v
+                else:
+                    prediction_filled_nans[k] = v.where(v.notnull(), existing_value)
```

Comment:
I think the implementation is correct, but it's hard to know if the broader integration with taper_nan_mask is working as you intended. You said there was some issue with the diagnostics that prevented you from saying for sure?

Reply:
Agreed the integration is trickier to test. I did verify offline that an 'out of sample'-wrapped tendency predictor give NaNs where expected. 

The monitored tendencies due to `fv3_physics`, `machine_learning` and `applied_physics` were not what I expected when plotted in a notebook. I'll make something more coherent to share.